tf2_ros/src/transform_listener.cpp

```diff
@@ -65,12 +66,13 @@ void TransformListener::initThread(
   rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base_interface)
 {
   using_dedicated_thread_ = true;
+  stop_thread_ = false;
   // This lambda is required because `std::thread` cannot infer the correct
   // rclcpp::spin, since there are more than one versions of it (overloaded).
   // see: http://stackoverflow.com/a/27389714/671658
   // I (wjwwood) chose to use the lamda rather than the static cast solution.
-  auto run_func = [](rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base_interface) {
-      return rclcpp::spin(node_base_interface);
+  auto run_func = [&](rclcpp::node_interfaces::NodeBaseInterface::SharedPtr node_base_interface) {
+      while (!stop_thread_) {rclcpp::spin_some(node_base_interface);}
```

Comment:
This is not a sufficient check. It currently neglects the condition on rclcpp shutdown.
It needs to be `while (!stop_thread_ && rclcpp::ok()) {....}`

with that change incorporated, the test doesn't fail anymore on my machine.